src/header.wppl

```diff
@@ -469,11 +469,47 @@ var OptimizeThenSample = function(wpplFn, options) {
   return SampleGuide(wpplFn, opts);
 };
 
-var Infer = function(options, maybeFn) {
-  if (!util.isObject(options)) {
-    util.fatal('Infer: expected first argument to be an options object.');
+/*
+ * DefaultInfer() called when no options are specified
+ * var maxEnumTreeSize: upper bound for enumeration tree size, enumeration ends when above threshold.
+ * var minSampleRate: lower bound for #sample/sec, rejection ends when below threshold.
+ * var samples: number of samples/particles used for default rejection/SMC/MCMC.
+ * next step in the heuristics flowchart
+ */
+var DefaultInfer = function(wpplFn, options) {
+  var maxEnumTreeSize = 200000;
+  var minSampleRate = 250; //from 1 to samples
+  var samples = 1000;
+  var enumResult = Enumerate(wpplFn, {maxEnumTreeSize: maxEnumTreeSize, maxRuntimeInMilliseconds: 5000,
+    throwOnError: false, strategy: 'depthFirst'});
+  if (dists.isDist(enumResult)) {
+    return enumResult;
+  } else {
```

Comment:
Here and below, I'd check that the inference result is an error object (`enumResult instanceof Error`) and throw an error if it's neither a distribution nor an error object.